File: Jaewon/Level.1/20) 시저 암호/20.js
function solution(s, n) {
    var answer = '';
    var lower = "abcdefghijklmnopqrstuvwxyz";
    var upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    
    let idx = 0;
    for(let i = 0; i < s.length; i++){
        if(s[i] === ' '){
            answer += ' ';
            continue;
        }
        if(lower.indexOf(s[i]) === -1){
            idx = upper.indexOf(s[i]) + n;
            idx = idx % 26;
            answer += upper[idx];
        } else {
            idx = lower.indexOf(s[i]) + n;
            idx = idx % 26;
            answer += lower[idx];
       }
    }
    return answer;
}

File: Jaewon/Level.1/20) 시저 암호/20.cpp
#include <string>
#include <vector>

using namespace std;

const string LOWER = "abcdefghijklmnopqrstuvwxyz";
const string UPPER = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";

string solution(string s, int n) {
    string answer = "";
    
    for(int i = 0; i < s.length(); i++){
        if(s[i] == ' '){
            answer += ' ';
            continue;
        }
        string textArr = islower(s[i]) == 0 ? UPPER : LOWER;
        int idx = textArr.find(s[i]) + n;
        if (idx >= textArr.length()) idx = idx % textArr.length();
        answer += textArr[idx];
    }
    return answer;
}